### PR TITLE
Handle previous Github comments

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Some comments from #314 were forgotten, they're implemented here.

Depends on #316

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt [hunk 1]
```diff
@@ -87,36 +89,49 @@ object DraftController {
...
+            when (draftMode) {
+                DraftMode.REPLY, DraftMode.REPLY_ALL -> draft.inReplyToUid = previousMessageUid
+                DraftMode.FORWARD -> draft.forwardedUid = previousMessageUid
+                DraftMode.NEW_MAIL -> Unit
             }
+
+            draft.to = previousMessage.from
+            if (draftMode == DraftMode.REPLY_ALL) draft.cc = previousMessage.to.union(previousMessage.cc).toRealmList()
+            previousMessage.subject?.let { draft.subject = "Re: $it" }
         }
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt:135]: All this code is executed if `previousMessageUid` is not null
So we might as well put it directly as non-null in the method to simplify everything

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt [hunk 2]
```diff
@@ -87,36 +89,49 @@ object DraftController {
         }?.localUuid
     }
 
-    fun MutableRealm.setDraftSignature(draftLocalUuid: String) {
-        updateDraft(draftLocalUuid, this) { draft ->
+    fun MutableRealm.setSignature(draft: Draft) {
 
-            draft.mimeType = ClipDescription.MIMETYPE_TEXT_HTML
+        draft.mimeType = ClipDescription.MIMETYPE_TEXT_HTML
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt:94]: Why not init it directly in the constructor?

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt [hunk 3]
```diff
@@ -87,36 +89,49 @@ object DraftController {
...
+                this.email = defaultSignature.replyTo
+                this.name = ""
+            })
+
+            val html = "<br/><br/><div class=\"editorUserSignature\">${defaultSignature.content}</div>"
+            draft.body = when (defaultSignature.position) {
+                SignaturePosition.AFTER_REPLY_MESSAGE -> draft.body + html
+                else -> html + draft.body
             }
+        }
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt:115]: I would see it rather in a method init of Draft, at its creation we give it a signature

**Review by JorisBodin — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt:140]: Done.

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt:99]: Done.

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt (hunk 3 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt]: This function has been moved to the `Draft` object.

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**